File: go_version/client/mainWindow/mainWindow.go
package mainWindow

import (
	"commsys/client/loginWindow"
	"commsys/client/registerWindow"
	"fmt"
	"os"
)

type MainWindow struct {
	Key  int
	Loop bool
	Lw   *loginWindow.LoginWindow
	Rw   *registerWindow.RegisterWindow
}

func (mw *MainWindow) MainShow() {
	// 接收用户的选择
	// 判断是否还继续显示菜单
	// fmt.Println("here")
	for mw.Loop {
		fmt.Println("--------------------欢迎登陆多人聊天系统--------------------")
		fmt.Println("                    1. 登录聊天室")
		fmt.Println("                    2. 注册用户")
		fmt.Println("                    3. 退出系统")
		fmt.Println("                    请选择（1-3）")

		fmt.Scanf("%d\n", &mw.Key)
		switch mw.Key {
		case 1:
			fmt.Println("登录聊天室")
			mw.Lw = &loginWindow.LoginWindow{}
			mw.Loop = false
		case 2:
			fmt.Println("注册用户")
			mw.Rw = &registerWindow.RegisterWindow{}
			mw.Loop = false
		case 3:
			fmt.Println("退出系统")
			mw.Loop = false
			os.Exit(0)
		default:
			fmt.Println("你的输入有误，请重新输入")
		}
	}

	if mw.Key == 1 {
		// 用户登录
		mw.Lw.LoginShow()
	} else if mw.Key == 2 {
		// 用户注册
		mw.Rw.ShowMenu()
		// fmt.Println("进行注册判断的逻辑")
	}
}

File: cpp_version/src/client/mainWindow.cpp

#include "../include/mainWindow.h"
#include <iostream>

void MainWindow::Show(){
    while (this->Loop) {
        std::cout <<"--------------------欢迎登陆多人聊天系统--------------------" << std::endl;
	    std::cout <<"                    1. 登录聊天室"<< std::endl;
	    std::cout <<"                    2. 注册用户"<< std::endl;
	    std::cout <<"                    3. 退出系统"<< std::endl;
	    std::cout <<"                    请选择（1-3）" << std::endl;
        std::cin >> this->Key;
        switch (this->Key) {
            case 1:
                std::cout << "登录聊天室" << std::endl;
                this->lw = new LoginWindow();
                this->Loop = false;
                break;
            case 2:
                std::cout << "注册用户" << std::endl;
                this->rw = new RegisterWindow();
                this->Loop = false;
                break;
            case 3:
                std::cout << "退出系统" << std::endl;
                this->Loop = false;
                exit(0);
            default:
                std::cout << "你的输入有误，请重新输入" << std::endl;
        }
    }

    if (this->Key == 1) {
    // 用户登录
	    this->lw->LoginShow();
    } else if (this->Key == 2) {
	    std::cout << "进行注册判断的逻辑" << std::endl;
        this->rw->ShowMenu();
	}
}